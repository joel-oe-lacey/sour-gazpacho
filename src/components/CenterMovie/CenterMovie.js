import React from 'react';
import './CenterMovie.scss';
import { Link } from 'react-router-dom';


const CenterMovie = ({ movie }) => {
  return (
    <article className="center-card">
      <div className='image-reflection'>
        <Link to={`/movies/${movie.id}`}><img src={movie.poster_path} alt='movie poster' className='center-poster'/></Link>
        <img src={movie.poster_path} alt='movie poster' className='center-reflection'/>
      </div>
      <div className='center-info'>
      </div>
      <section className='text-box'>
        <h3 className='center-text' id='center-title'>{movie.title}</h3>
        <h4 className='center-text' id='center-rating'>Average Rating: {(movie.average_rating).toFixed(1)}</h4>
        <h4 className='center-text' id='center-date'>{month} {formatDate[2]}, {formatDate[0]}</h4>
        <p className='center-text' id='center-overview'>{movie.overview}</p>
      </section>
    </article>
  )
}

export default CenterMovie
