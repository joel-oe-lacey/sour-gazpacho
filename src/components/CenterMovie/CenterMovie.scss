@import '../../variables';

.center-card {
  width: 440px;
  height: 540px;
  margin: 20px;
  color: rgba(0,0,0,0);
  display: flex;
  flex-direction: row;
  margin-left: -100px;
}

.center-reflection {
  display: flex;
  flex-direction: column;
}

.center-poster {
  animation: fade-in 6s cubic-bezier(0.390, 0.575, 0.565, 1.000) infinite both;
  border: solid 5px $primary-red-color;
  margin-top: 20px;
  height: 500px;
  width: 400px;
  object-fit: cover;
  z-index: 10;
  transform: perspective(500px) rotateY(10deg);
}

.center-reflection {
  animation: fade-in 6s cubic-bezier(0.390, 0.575, 0.565, 1.000) infinite both;
  border: solid 5px $primary-red-color;
  object-fit: cover;
  height: 500px;
  width: 403px;
  transform: scale(1,-1) perspective(500px) rotateY(-10deg);
  opacity: 0.3;
  margin-top: 10px;
  margin-left: -30px;
  mask-image: linear-gradient(to top, rgba(0,0,0,0.5) 0%, rgba(0,0,0,0) 25%);
}

@keyframes fade-in {
  0% {
    opacity: 0;
  }
  10% {
    opacity: 1;
  }
  85% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}

.text-box {
  display: flex;
  flex-direction: column;
}

.center-text {
  animation: fade-in 6s cubic-bezier(0.390, 0.575, 0.565, 1.000) infinite both;
  color: #FFF;
  font-size: 20px;
  width: 500px;
  margin-left: 30px;
}

#center-title {
  font-size: 60px;
  margin-bottom: 50px;
  margin-top: 50px;
}

#center-rating {
  color: $primary-red-color;
  font-size: 30px;
  margin-bottom: 20px;
  margin-top: 20px;
}

#center-date {
  font-size: 30px;
  margin-bottom: 20px;
  margin-top: 20px;
}

#center-overview {
  font-size: 20px;

}
