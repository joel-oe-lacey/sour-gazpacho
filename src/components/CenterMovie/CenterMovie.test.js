import React from 'react';
import CenterMovie from './CenterMovie.js';
import { shallow } from 'enzyme';
import { mockRandom } from 'jest-mock-random';

describe('CenterMovie', () => {

  beforeEach(() => {
    mockRandom([0.1]);
  })

  it('should match the snapshot', () => {
    let wrapper = shallow(<CenterMovie
      movie={[{
        title:'Parasite',
        poster_path:'https:/image.some.jpg',
        average_rating:5,
        overview:'Movie Summary',
        release_date:'2020-01-12'
      }, {
        title:'Sonic The Hedhehog',
        poster_path:'https:/image.some.jpg',
        average_rating:7,
        overview:'Movie Summary',
        release_date:'2020-10-19'
      }, {
        title:'Charlie\'s Angels',
        poster_path:'https:/image.some.jpg',
        average_rating:7,
        overview:'Movie Summary',
        release_date:'2019-12-10'
      }]}
      />);
      let mockState = {
        currentMovie: {
          title:'Sonic The Hedhehog',
          poster_path:'https:/image.some.jpg',
          average_rating:7,
          overview:'Movie Summary',
          release_date:'2020-10-19'
        }
      }
    wrapper.setState(mockState);
    expect(wrapper).toMatchSnapshot();
  })

  it('should setState with a movie when component mounts', () => {
    let wrapper = shallow(<CenterMovie
      movie={[{
        title:'Parasite',
        poster_path:'https:/image.some.jpg',
        average_rating:5,
        overview:'Movie Summary',
        release_date:'2020-01-12'
      }, {
        title:'Sonic The Hedhehog',
        poster_path:'https:/image.some.jpg',
        average_rating:7,
        overview:'Movie Summary',
        release_date:'2020-10-19'
      }, {
        title:'Charlie\'s Angels',
        poster_path:'https:/image.some.jpg',
        average_rating:7,
        overview:'Movie Summary',
        release_date:'2019-12-10'
      }]}
      />);
    const expected = {
      title:'Charlie\'s Angels',
      poster_path:'https:/image.some.jpg',
      average_rating:7,
      overview:'Movie Summary',
      release_date:'2019-12-10'
    };
    wrapper.instance().componentDidMount();
    expect(wrapper.state('currentMovie')).toEqual(expected);
  })

})
