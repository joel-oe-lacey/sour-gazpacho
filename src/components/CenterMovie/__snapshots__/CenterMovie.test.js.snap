// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterMovie should match the snapshot 1`] = `
<article
  className="center-card"
>
  <div
    className="image-reflection"
  >
    <Link
      to="/movies/undefined"
    >
      <img
        alt="movie poster"
        className="center-poster"
        src="https:/image.some.jpg"
      />
    </Link>
    <img
      alt="movie poster"
      className="center-reflection"
      src="https:/image.some.jpg"
    />
  </div>
  <div
    className="center-info"
  />
  <section
    className="text-box"
  >
    <h3
      className="center-text"
      id="center-title"
    >
      Parasite
    </h3>
    <h4
      className="center-text"
      id="center-rating"
    >
      Average Rating: 
      5.0
    </h4>
    <h4
      className="center-text"
      id="center-date"
    >
      January
       
      12
      , 
      2020
    </h4>
    <p
      className="center-text"
      id="center-overview"
    >
      Movie Summary
    </p>
  </section>
</article>
`;
